_pydevd_sys_monitoring/_pydevd_sys_monitoring.py

```diff
@@ -897,15 +922,23 @@ def _raise_event(code, instruction, exc):
 
     # print('_raise_event --- ', code, exc)
 
-    frame = _getframe(1)
-    arg = (type(exc), exc, exc.__traceback__)
+    # Compute the previous exception info (if any). We use it to check if the exception
+    # should be stopped
+    prev_exc_info = _thread_local_info._user_uncaught_exc_info if hasattr(_thread_local_info, "_user_uncaught_exc_info") else None
     should_stop, frame, _user_uncaught_exc_info = should_stop_on_exception(
-        py_db, thread_info.additional_info, frame, thread_info.thread, arg, None
+        py_db, thread_info.additional_info, frame, thread_info.thread, arg, prev_exc_info
     )
+
+    # Save the current exception info for the next raise event.
+    _thread_local_info._user_uncaught_exc_info = _user_uncaught_exc_info
+
     # print('!!!! should_stop (in raise)', should_stop)
     if should_stop:
         handle_exception(py_db, thread_info.thread, frame, arg, EXCEPTION_TYPE_HANDLED)
-        return
+
+    # Once we leave the raise event, we are no longer in the state of 'just_raised', so 
+    # indicate that this traceback is for an exception in the unwinding state
+    flag_as_unwinding(exc.__traceback__)
```

Comment:
The django plugin was messing up when trying to determine if an exception was just raised or not. This special flag makes that work.

Reply:
I'll create a test that matches the one in debugpy. I believe this is the one that was failing:
https://github.com/microsoft/debugpy/blob/25955a05d87a6e3d7694f7479c1c17ded0f9af90/tests/debugpy/test_django.py#L138

It was in the django plugin where it was trying to determine if a template exception was being thrown I believe.

Reply:
Updated your test in test_debugger_json that validates this change. Without it the test fails.

Reply:
I changed the logic though to pass through whether or not we're 'unwinding' instead of holding onto the traceback